### DISCUSSION TITLE
revalidateTag without incrementalCacheHandlerPath

### DISCUSSION THREAD

**Discussion by landvibe** [vercel/next.js#55902] [Help]
### Summary

I understand that the default cache handler uses the local filesystem. Since the revalidate attribute of fetch revalidates based on time, it seems fine without implementing incrementalCacheHandlerPath. In that case, is it safe to use revalidateTag without implementing incrementalCacheHandlerPath (with the default cache handler)?

Using revalidateTag with the default cache handler seems to revalidate only the cache of a single instance, which may lead to the following situation:

1. User 1: Connects to Instance 1 and revalidates the data cache through revalidateTag.
2. Instance 1’s data cache is revalidated.
3. User 2: Connects to Instance 2. It uses outdated data.

I would like to know if there are any reference materials for implementing incrementalCacheHandlerPath on AWS.

**Comment by better-salmon** [comment]
Hey @landvibe, check out https://caching-tools.github.io/next-shared-cache/. This library supports the App and Pages routers and provides pre-configured cache handlers with built-in revalidation for multi-instance environments, along with guidance on implementing your own cache handler with revalidation support.
